### PR TITLE
Restaurant Geolocation

### PR DESCRIPTION
**Pull request by aljoh**
### PT Link: https://www.pivotaltracker.com/story/show/154174433

### CODE REVIEW

File: config/initializers/geocoder.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Geocoder.configure(
+  timeout: 3,
+  lookup: :google,
+  api_key: ENV['GEOCODER_APIKEY'],
```
Comment (tochman) [inline config/initializers/geocoder.rb]: Where is this ENV variable set?

File: spec/models/restaurant_spec.rb [hunk 2]
```diff
@@ -3,10 +3,18 @@
...
+    it { is_expected.to have_db_column :street_address }
+    it { is_expected.to have_db_column :longitude }
+    it { is_expected.to have_db_column :latitude }
   end
 
   describe 'Validations' do
     it { is_expected.to validate_presence_of :name }
+    it { is_expected.to validate_presence_of :city }
+    it { is_expected.to validate_presence_of :post_code }
+    it { is_expected.to validate_presence_of :street_address }
```
Comment (tochman) [inline spec/models/restaurant_spec.rb]: I would like you to include a spec for `#full_address` method. At the very least a `responds_to` spec

**Review by tochman — CHANGES_REQUESTED** [review]
IMO this should have been a feature, not a chore. I'm missing acceptance tests. Not a blocker though.  

Do address my comments before merge.

File: config/initializers/geocoder.rb (hunk 1 above)
Comment (aljoh) [reply config/initializers/geocoder.rb]: In a hidden .env file

File: spec/models/restaurant_spec.rb (hunk 2 above)
Comment (aljoh) [reply spec/models/restaurant_spec.rb]: Spec included, not sure if included in a good way though

File: config/initializers/geocoder.rb (hunk 1 above)
Comment (tochman) [reply config/initializers/geocoder.rb]: Where? I see no dotenv gem

File: spec/models/restaurant_spec.rb (hunk 2 above)
Comment (tochman) [reply spec/models/restaurant_spec.rb]: I see no spec for full_address.

Comment (aljoh) [reply spec/models/restaurant_spec.rb]: At the bottom of the file

File: config/initializers/geocoder.rb (hunk 1 above)
Comment (aljoh) [reply config/initializers/geocoder.rb]: Yeah it wasn't added.. my bad :) Up and running now

File: spec/models/restaurant_spec.rb (hunk 2 above)
Comment (tochman) [reply spec/models/restaurant_spec.rb]: Alfred, that is `street_address` right? I'm asking for unit tests for the `full_address` method you have on your model. Please read my comments carefully.

File: spec/models/restaurant_spec.rb [hunk 3]
```diff
@@ -22,4 +22,11 @@
       expect(FactoryBot.create(:restaurant)).to be_valid
     end
   end
+
+  before do
+    FactoryBot.create(:restaurant)
```
Comment (tochman) [inline spec/models/restaurant_spec.rb]: What does that do for you? Where do you use this object?

**Review by tochman — CHANGES_REQUESTED** [review]

File: spec/models/restaurant_spec.rb [hunk 4]
```diff
@@ -3,15 +3,30 @@
...
     it 'should be valid' do
       expect(FactoryBot.create(:restaurant)).to be_valid
     end
   end
+
+  before do
+    FactoryBot.create(:restaurant)
+  end
+  it 'responds to full_address' do
+    expect(Restaurant.first).to respond_to(:full_address)
```
Comment (aljoh) [inline spec/models/restaurant_spec.rb]: here @tochman

Comment (tochman) [reply spec/models/restaurant_spec.rb]: Okay. That is a very basic test. Thank you

Comment (tochman) [reply spec/models/restaurant_spec.rb]: If you are satisfied with this standard of testing rhen I will not interfere.

Comment (oliverochman) [reply spec/models/restaurant_spec.rb]: Any suggestions on how to improve the test?

Comment (tochman) [reply spec/models/restaurant_spec.rb]: Yes, take a look at betterspecs.org. 

You know, use `describe`, follow a standard (like referencing method names tith `#`), check for the actual values of `full_address`, etc...)

File: spec/models/restaurant_spec.rb (hunk 3 above)
Comment (tochman) [reply spec/models/restaurant_spec.rb]: I still need a clarification on this before block.

**Review by zuzanna81 — COMMENTED** [review]
Remove commented lines in db/schema.rb

**Comment by diraulo** [comment]
Update this Branch with upstream's latest code

**Review by tochman — APPROVED** [review]

**Pull request merged**